server/neptune/workflows/internal/deploy/terraform/state.go

```diff
@@ -39,6 +48,35 @@ func (n *StateReceiver) Receive(ctx workflow.Context, c workflow.ReceiveChannel,
 		}
 	}
 
+	if workflowState.Apply != nil &&
+		workflowState.Apply.Status == state.WaitingJobStatus &&
+		!reflect.ValueOf(workflowState.Apply.OnWaitingActions).IsZero() {
+		// update queue with information about current deployment pending confirm/reject action
+		infos := n.Queue.GetOrderedMergedItems()
+
+		revisionsSummary := n.Queue.GetQueuedRevisionsSummary()
+		state := github.CheckRunQueued
+		runLink := github.BuildRunURLMarkdown(deploymentInfo.Repo.GetFullName(), deploymentInfo.Commit.Revision, deploymentInfo.CheckRunID)
+		summary := fmt.Sprintf("This deploy is queued pending action on run for revision %s.\n%s", runLink, revisionsSummary)
+
+		for _, i := range infos {
+			request := notifier.GithubCheckRunRequest{
```

Comment:
do these checkruns conflict with the checkruns managed by the terraform state notifier?  what are your thoughts on placing this logic within the terraform state notifier?  we can probably add some generic fields in the `internalTerraform.DeploymentInfo` to capture other deployments that are relevant, in order to avoid a circular dependency